### PR TITLE
User's codes feed FE

### PR DESCRIPTION
**Pull request by dayanamdr**

### CODE REVIEW

File: codewell-fe/src/Code/UserFeed.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import 'antd/dist/antd.css';
+import { List, Layout } from 'antd';
+
+function UserCodeFeed() {
+  const [userCodeFeed, setUserCodeFeed] = useState([]);
+  const [error, setError] = useState("");
+  let { username } = useParams();
+
+  const contentStyle = {textAlign: 'center'};
+  const listStyle = {
```
Comment (andreisamoila74) [inline codewell-fe/src/Code/UserFeed.js:16]: why do we need this?

File: codewell-fe/src/Code/UserFeed.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    width: '40%',
+    border: '1px solid #bfbfbf',
+    backgroundColor: '#ffffff'
+  };
+  const pageStyle = {alignItems: 'center'};
```
Comment (andreisamoila74) [inline codewell-fe/src/Code/UserFeed.js:21]:
```suggestion
  const pageStyle = { alignItems: 'center' };
```

File: codewell-fe/src/Code/UserFeed.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          setError(data.error);
+            console.log(data.error);
+        }
+      })
+  }, []);
+
+  return (
+    <>
+    {error
+      ? <PermissionsError error={error} />
```
Comment (andreisamoila74) [inline codewell-fe/src/Code/UserFeed.js:38]: you don't need to pass any error here

**Review by andreisamoila74 — CHANGES_REQUESTED** [review]

File: codewell-fe/src/Code/Feed.js [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
 import { List, Layout } from 'antd';
 
 function CodeFeed() {
   const [codeFeed, setCodeFeed] = useState([]);
+  const [error, setError] = useState("");
+  let { userId } = useParams();
+
+  if (userId === undefined) {
+    userId = '';
+  }
```
Comment (andreisamoila74) [inline codewell-fe/src/Code/Feed.js:15]:
```suggestion
 userId = userId === undefined ? '' : userId;
```

File: codewell-fe/src/__tests__/Code/Feed.test.js [hunk 5]
```diff
@@ -1,47 +1,215 @@
-import React from 'react';
+import React from 'react'
```
Comment (andreisamoila74) [inline codewell-fe/src/__tests__/Code/Feed.test.js:1]: please don't remove ;

File: codewell-fe/src/__tests__/Code/Feed.test.js [hunk 6]
```diff
@@ -1,47 +1,215 @@
...
 
-import CodeFeed from '../../Code/Feed.js';
-import { render, wait } from '@testing-library/react';
-import AdminRoute from '../../Permissions/AdminRoute.js';
-import { MemoryRouter } from 'react-router-dom';
+import CodeFeed from '../../Code/Feed.js'
```
Comment (andreisamoila74) [inline codewell-fe/src/__tests__/Code/Feed.test.js:3]:
```suggestion
import CodeFeed from '../../Code/Feed.js';
```

File: codewell-fe/src/__tests__/Code/Feed.test.js [hunk 7]
```diff
@@ -1,47 +1,215 @@
...
+      </MemoryRouter>
+    );
+
+    await wait(() => expect(global.fetch).toHaveBeenCalled());
+    expect(container.textContent)
+      .toContain('You are not authorized to access this page!');
+  }
+);
+
+test('Render session user code feed if non admin user requests the code feed',
```
Comment (andreisamoila74) [inline codewell-fe/src/__tests__/Code/Feed.test.js:157]: i don't understand this, can you explain please?

**Review by andreisamoila74 — CHANGES_REQUESTED** [review]
we don't need package-lock.json modified

**Review by andreisamoila74 — APPROVED** [review]

File: codewell-fe/src/Code/Feed.js [hunk 8]
```diff
@@ -15,32 +20,39 @@ function CodeFeed() {
   };
   const pageStyle = {alignItems: 'center'};
   useEffect(() => {
-    fetch('/api/code/feed')
+    fetch('/api/code/feed?user_id=' + userId)
```
Comment (S7012MY) [inline codewell-fe/src/Code/Feed.js:23]: can we keep the back end consistent with the front end /feed/user_id?

**Review by S7012MY — CHANGES_REQUESTED** [review]

**Review by S7012MY — APPROVED** [review]

**Pull request merged**